test/consts/exit_codes.go

```diff
@@ -0,0 +1,35 @@
+package consts
+
+// -----------------------------------------------------------------------------
+// Test Suite Exit Codes
+// -----------------------------------------------------------------------------
+
+const (
+	// ExitCodeIncompatibleOptions is a POSIX compliant exit code for the test suite to
+	// indicate that some combination of provided configurations were not compatible.
+	ExitCodeIncompatibleOptions = 100
+
+	// ExitCodeInvalidOptions is a POSIX compliant exit code for the test suite to indicate
+	// that some of the provided runtime options were not valid and the tests could not run.
+	ExitCodeInvalidOptions = 101
+
+	// ExitCodeCantUseExistingCluster is a POSIX compliant exit code for the test suite to
+	// indicate that an existing cluster provided for the tests was not usable.
+	ExitCodeCantUseExistingCluster = 101
+
+	// ExitCodeCantCreateCluster is a POSIX compliant exit code for the test suite to indicate
+	// that a failure occurred when trying to create a Kubernetes cluster to run the tests.
+	ExitCodeCantCreateCluster = 102
+
+	// ExitCodeCleanupFailed is a POSIX compliant exit code for the test suite to indicate
+	// that a failure occurred during cluster cleanup.
+	ExitCodeCleanupFailed = 103
+
+	// ExitCodeEnvSetupFailed is a generic exit code that can be used as a fallback for general
+	// problems setting up the testing environment and/or cluster.
+	ExitCodeEnvSetupFailed = 104
+
+	// ExitCodeCentCreateLogger is a POSIX compliant exit code for the test suite to indicate
+	// that a failure occurred when trying to create a logger for the test suite.
+	ExitCodeCantCreateLogger = 105
+)
```

Comment:
Do you have any idea how these specific exit codes are meant to help us? As I can see, the `exitOnErrWithCode` function that they are used in pair with, logs the error to stderr, which should be visible in the CI. I don't understand how using specific error codes will give us more context than the error log. 🤔

Reply:
I understand it was just moved so don't bother to change it in this PR, but I'm just curious if I'm missing something.

Reply:
Perhaps to be POSIX compliant just like mentioned in the comments :)

I believe these are a nice touch for someone running the tests and inspecting the return code programmatically 🤷